<launch>
  <node pkg="image_view2" type="image_view2" name="image_view2">
    <remap from="image" to="/multisense/left/image_rect_color" />
  </node>
  <node pkg="jsk_perception" type="rect_to_mask_image" name="mask_image">
    <remap from="~input" to="/multisense/left/image_rect_color/screenrectangle" />
    <remap from="~input/camera_info" to="/multisense/left/camera_info" />
  </node>
  <node pkg="jsk_pcl_ros" type="mask_image_filter" name="mask_image_filter">
    <remap from="~input" to="/multisense/resize_1_4/points" />
    <remap from="~input/mask" to="mask_image/output" />
    <remap from="~input/camera_info" to="/multisense/left/camera_info" />
  </node>
  <node pkg="nodelet" type="nodelet" name="mask_image_filter_points"
        args="standalone pcl/ExtractIndices">
    <remap from="~input" to="/multisense/resize_1_4/points" />
    <remap from="~indices" to="mask_image_filter/output" />
  </node>
  <node pkg="jsk_pcl_ros" type="tf_transform_cloud" name="ground_cloud">
    <remap from="~input" to="mask_image_filter_points/output" />
    <rosparam>
      target_frame_id: ground
    </rosparam>
  </node>
  <node pkg="jsk_pcl_ros" type="belief_cloud_filter" name="belief_cloud_filter">
    <remap from="~input" to="ground_cloud/output" />
    <rosparam>
      initial_model: gaussian
      gaussian_means: [2, 0, 1.2]
      gaussian_covariances: [1.0, 1.0, 0.2]
    </rosparam>
  </node>
</launch>
